SliderButton/SliderButtonBehaviorDecorator.js

```diff
@@ -10,43 +12,71 @@ import {findDOMNode} from 'react-dom';
  * @memberof agate/SliderButton
  * @private
  */
-const SliderButtonBehaviorDecorator = (Wrapped) => {
-	// eslint-disable-next-line no-shadow
-	function SliderButtonBehaviorDecorator ({onChange, ...rest}) {
-		const {children} = rest;
-		const [valueText, setValueText] = React.useState(children ? children[0] : null);
-		const ref = React.useRef();
-
-		function handleChange ({value}) {
-			onChange();
-			setValueText(children[value]);
+const SliderButtonBehaviorDecorator = hoc((config, Wrapped) => {
+	return class extends React.Component {
+		static displayName = 'SliderButtonBehaviorDecorator';
+
+		static propTypes = /** @lends agate/ArcSlider.ArcSliderBehaviorDecorator.prototype */ {
+			/**
+			 * Items displayed with SliderButton.
+			 *
+			 * @type {Function}
+			 * @private
+			 */
+			children: PropTypes.node,
+
+			/**
+			 * Called when value is changed.
+			 *
+			 * @type {Function}
+			 * @private
+			 */
+			onChange: PropTypes.func
+		};
+
+		constructor (props) {
+			super(props);
+
+			this.ref = React.createRef();
+
+			this.state = {
+				valueText: props.children ? props.children[0] : null
+			};
 		}
 
-		function handleDragStart () {
+		handleChange = ({value}) => {
+			this.setState(
+				() => ({valueText: this.props.children[value]}),
+				() => {
+					forward('onChange', {
+						type: 'onChange',
+						value
+					}, this.props);
+				}
+			);
+		};
```

Comment:
Is there any reason to call `forward` async? In other components, we just called it sync. So I'm curious about the needs for the async call.
```suggestion
		handleChange = ({value}) => {
			this.setState(() => ({valueText: this.props.children[value]}));
			forward('onChange', {
				type: 'onChange',
				value
			}, this.props);
		};
```

Reply:
No reason as well. I changed it to call it sync. Thanks!